src/controller/CHIPDeviceController.cpp

```diff
@@ -958,6 +958,15 @@ void DeviceCommissioner::RendezvousCleanup(CHIP_ERROR status)
 {
     FreeRendezvousSession();
 
+    if (mDeviceBeingCommissioned != nullptr)
```

Comment:
This code block was causing issues for network commissioning. It was removed in this PR: https://github.com/project-chip/connectedhomeip/pull/11592
Just wanted to point out to make sure network commissioning is working with the current PR.

Reply:
@cecille ^

Reply:
Yeah, the original DeviceProxy PR changed this to be called before commissioning was actually done and AdvanceCommissioningStage relied on mDeviceBeingCommissioned to know where to send the commissioning complete command. The hotfix changed that to just keep the mDeviceBeingCommissioned around, and sent the commissioning compete command over the CommissioneeDeviceProxy. That's not totally correct, because that command is meant to be sent over the operational network (previously this was swapped under the hood so to speak). Now we explicitly grab and store the operational DeviceProxy in OnDeviceConnectedFn and use that the send the commissioning complete command. 

I tested this using chip-tool and the lighting app, and on-network commissioning completes fine. Also, chip-tool is now updated to explicitly look for the commissioning complete call - before it was just calling it good after any callback, which was probably why this didn't get detected by the tests (and also what was causing so many weird failure on this PR).